### PR TITLE
[ADD] account_analytic_report: New module account_analytic_report

### PR DESCRIPTION
**Pull request by BernatObrador**
New module account_analytic_report

Using this module is straightforward. Follow these steps:

* **Navigate to the Report**:  
  Go to **Invoicing** -> **Reporting** -> **Analytic Trial Balance**.

* **Customize the Report with Filters**:  
  Adjust the report using the available options:

  * **Group by Analytic Account**:  
    Groups the results by analytic accounts instead of financial accounts.

  * **Show Hierarchy and Limit Hierarchy Level**:  
    Displays the amounts split by the hierarchy levels of financial accounts.

  * **Filter Accounts**:  
    When used independently (without grouping by analytic accounts or showing hierarchy), the results will be split by both financial accounts.  
    **Example**: Filtering by accounts *Test 1* and *Test 2*:

    ```text
            | Initial Balance | Test 1   | Test 2   | Ending Balance
    400000  |        0        | $3600    | $2400    |     $6000
    ```

  * **Show Months** (Excel export only):  
    Enabled when filtering accounts without grouping by analytic accounts or showing hierarchy. It generates a separate sheet in the Excel file for each filtered account, detailing the amounts by month within the selected date range.  

cc https://github.com/APSL 162352

@miquelalzanillas @lbarry-apsl @mpascuall @peluko00 @javierobcn @ppyczko please review

### CODE REVIEW

File: account_analytic_report/wizard/trial_balance_analytic_wizard_view.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                <div>
+                    <group name="filters">
+                        <group name="date_range">
+                            <field name="date_range_id" />
+                            <field name="date_from" required="1" />
+                            <field name="date_to" required="1" />
+                            <field name="fy_start_date" invisible="1" />
+                        </group>
+                        <group name="other_filters">
+                            <field name="plan_id" required="1" />
```
Comment (peluko00) [inline account_analytic_report/wizard/trial_balance_analytic_wizard_view.xml]: Put required on a declaration of the field

**Comment by BernatObrador** [comment]
Thanks for your comment @peluko00! Could you review again please?

**Review by peluko00 — APPROVED** [review]
Works as expected in runboat and code review

**Review by mpascuall — APPROVED** [review]
LGTM, nice work!

**Comment by pedrobaeza** [comment]
Shouldn't this be in OCA/account-analytic instead?

**Comment by BernatObrador** [comment]
@pedrobaeza You are totally right. Here is the new pr: [#730](https://github.com/OCA/account-analytic/pull/730)

**Pull request closed without merging**